ang/test/civicase/activity/filters/directives/activity-filters.directive.spec.js

```diff
@@ -2,25 +2,49 @@
 
 (function ($, _) {
   describe('civicaseActivityFilters', function () {
-    var $compile, $rootScope, $scope, activityFilters;
+    var $compile, $rootScope, $scope, activityFilters, CaseTypeCategory;
 
     beforeEach(module('civicase', 'civicase.templates', function () {
       killDirective('civicaseActivityFiltersContact');
     }));
 
-    beforeEach(inject(function (_$compile_, _$rootScope_) {
+    beforeEach(inject(function (_$compile_, _$rootScope_, _CaseTypeCategory_) {
       $compile = _$compile_;
       $rootScope = _$rootScope_;
+      CaseTypeCategory = _CaseTypeCategory_;
       spyOn($rootScope, '$broadcast');
+      spyOn(CaseTypeCategory, 'getCategoriesWithAccessToActivity')
+        .and.returnValue([CaseTypeCategory.getAll()[1]]);
```

Comment:
We should put the mocked data in a variable that explains their nature instead of having `[CaseTypeCategory.getAll()[1]]`. Can we refactor this part like this:

```js
categoriesWhereUserCanAccessActivities = [CaseTypeCategory.getAll()[1]];

spyOn(CaseTypeCategory, 'getCategoriesWithAccessToActivity')
  .and.returnValue(categoriesWhereUserCanAccessActivities);
```

Also:

```js
categoriesWhereUserCanAccessActivities = _.sample(CaseTypeCategory.getAll(), 1);
```

It's a bit less intimidating than the current implementation since there are less brackets and parentheses to follow.